pygmt/tests/test_nearneighbor.py

```diff
@@ -0,0 +1,104 @@
+"""
+Tests for nearneighbor.
+"""
+import os
+
+import numpy.testing as npt
+import pytest
+import xarray as xr
+from pygmt import nearneighbor
+from pygmt.datasets import load_sample_bathymetry
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile, data_kind
+
+
+@pytest.fixture(scope="module", name="ship_data")
+def fixture_ship_data():
+    """
+    Load the data from the sample bathymetry dataset.
+    """
+    return load_sample_bathymetry()
+
+
```

Comment:
I think we can merge the `test_nearneighbor_input_file` and `test_nearneighbor_input_numpy_array` functions using `@pytest.mark.parametrize` decorator.

Reply:
Tests have been merged in 334939609e8d3b5533f4f81398237accba2d5bb7. It was hard to parametrize fixtures (see https://stackoverflow.com/questions/51439888/how-to-use-pytest-mark-parametrize-with-fixtures) so I modified the test slightly to check for `numpy.array`/`xarray.Dataset` inputs.